dronekit_sitl/__init__.py

```diff
@@ -137,14 +138,59 @@ def check_complete(count, block_size, total_size):
     if verbose:
         print('Ready to boot.')
 
+sitl_instance_count = 0
+
+class ArdupilotCapabilities():
```

Comment:
I think that would be over-engineering at this point.  Every option in there at the moment is quite ArudPilot specific (including the ones you mention); I don't think any of them would be directly relevant to any other autopilot!

I think this sort of change would be a good one to make when someone was trying to make dronekit-sitl work with another SITL system, as any commonality will be evident to them at that time.